apps/server/src/modules/files-storage/uc/files-storage.uc.spec.ts

```diff
@@ -199,374 +170,533 @@ describe('FilesStorageUC', () => {
+
+				const expectedFileDescription = IFileBuilder.buildFromAxiosResponse(uploadFromUrlParams.fileName, response);
+				expect(filesStorageService.uploadFile).toHaveBeenCalledWith(
+					userId,
+					uploadFromUrlParams,
+					expectedFileDescription
+				);
 			});
 
-			request.get.mockReturnValue('1234');
-			request.pipe.mockImplementation(mockBusboyEvent as never);
-		});
+			it('should call uploadFile get with correct params', async () => {
+				const { uploadFromUrlParams, userId, fileRecord } = setup();
 
-		it('should call request.get()', async () => {
-			await filesStorageUC.upload(userId, fileUploadParams, request);
-			expect(request.get).toBeCalledWith('content-length');
-			expect(request.get).toHaveBeenCalledTimes(1);
-		});
+				const result = await filesStorageUC.uploadFromUrl(userId, uploadFromUrlParams);
 
-		it('should call request.pipe()', async () => {
-			await filesStorageUC.upload(userId, fileUploadParams, request);
-			expect(request.pipe).toHaveBeenCalledTimes(1);
+				expect(result).toEqual(fileRecord);
+			});
 		});
 
-		it('should call fileRecordRepo.uploadFile', async () => {
-			await filesStorageUC.upload(userId, fileUploadParams, request);
-			expect(storageClient.create).toHaveBeenCalledTimes(1);
-		});
+		describe('WHEN user is not authorised', () => {
+			const setup = () => {
+				const { userId, uploadFromUrlParams } = getUploadFromUrlParams();
+				const error = new Error('test');
+				authorizationService.checkPermissionByReferences.mockRejectedValueOnce(error);
 
-		it('should call fileRecordRepo.uploadFile with params', async () => {
-			await filesStorageUC.upload(userId, fileUploadParams, request);
+				return { uploadFromUrlParams, userId, error };
+			};
 
-			const storagePath = [schoolId, entityId].join('/');
+			it('should pass error', async () => {
+				const { uploadFromUrlParams, userId, error } = setup();
 
-			expect(storageClient.create).toBeCalledWith(storagePath, {
-				buffer: Buffer.from('abc'),
-				name: 'text.txt',
-				size: 1234,
-				mimeType: 'text/plain',
+				await expect(filesStorageUC.uploadFromUrl(userId, uploadFromUrlParams)).rejects.toThrow(error);
 			});
 		});
 
-		it('should return instance of FileRecord', async () => {
-			const result = await filesStorageUC.upload(userId, fileUploadParams, request);
-			expect(result).toBeInstanceOf(FileRecord);
-		});
+		describe('WHEN httpService throws error', () => {
+			const setup = () => {
+				const { userId, uploadFromUrlParams } = getUploadFromUrlParams();
 
-		describe('save() with FileName Handling', () => {
-			it('should call fileRecordRepo.save', async () => {
-				await filesStorageUC.upload(userId, fileUploadParams, request);
-				expect(fileRecordRepo.save).toHaveBeenCalledTimes(1);
-			});
+				const error = createObservable({
+					isAxiosError: true,
+					code: '404',
+					response: {},
+					name: 'errorText',
+					message: 'errorText',
+					toJSON: () => ({}),
+				}) as never;
```

Comment:
"as never" feel wrong on any place where i see it.

Reply:
All those never typing issues should already come up in the ticket they where introduced. Do we want to solve it in this ticket now?

Reply:
Never is removed and correct axios error was added.